netsim/augment/nodes.py

```diff
@@ -334,6 +334,44 @@ def augment_node_device_data(n: Box, defaults: Box) -> None:
     log.error(f"Daemon configuration files for node {n} must be a dictionary")
     n.pop('_daemon_config',None)
 
+'''
+Set addresses of the main loopback interface
+'''
+def loopback_interface(n: Box, pools: Box, topology: Box) -> None:
+  n.loopback.type = 'loopback'
+  n.loopback.neighbors = []
+  n.loopback.virtual_interface = True
+  lbname = devices.get_loopback_name(n,topology)
+  if lbname:
+    n.loopback.ifname = lbname
+    n.loopback.ifindex = 0
+
+  pool = n.get('loopback.pool','loopback')
+  prefix_list = addressing.get(pools,[ pool ],n.id)
+
+  for af in prefix_list:
+    if prefix_list[af] is True:
+      log.error(
+        f"Address pool {pool} cannot contain unnumbered/LLA addresses",
+        category=log.IncorrectType,
+        module='nodes')
+    elif not n.loopback[af] and not (prefix_list[af] is False):
+      if af == 'ipv6':
+        if prefix_list[af].prefixlen == 128:
+          n.loopback[af] = str(prefix_list[af])
+        else:
+          n.loopback[af] = addressing.get_addr_mask(prefix_list[af],1)
```

Comment:
There was a little hack I had to make in order to get around a loopback issue: https://github.com/ipspace/netlab/blob/bccc52f0d250c0e4ecb8f16ca4b77732ac5296b5/netsim/ansible/templates/initial/sros.j2#L41

SR OS requires a /128 on its primary loopback. For consistency, it would be better if there were a device property expressing any loopback prefix lengths restrictions

Reply:
That sounds like a device quirk to me ;)

Reply:
What's wrong with setting **prefix6** in **addressing.loopback** (https://netlab.tools/addressing/#specifying-address-pools) or its corresponding system default?

Reply:
That didn't exist at the time I wrote that template - it was only introduced on October 16th 2023 https://github.com/ipspace/netlab/commit/4a6fd95c8e0c43bab9fc0c31d8be27d38798f114

Reply:
🤷‍♂️